src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -16,28 +19,97 @@
 
     public static final String COMMAND_WORD = "find";
 
+    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1d persons listed!";
+
+    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW_ATTRIBUTE = "%1d persons listed with %s attribute%s!";
+
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose chosen field contains any of "
-            + "the specified keywords (case-insensitive) and displays them as a list\n"
-            + "with index numbers.\n"
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Use a flag (n/, p/, e/, a/, t/, i/) to search by name, phone, email, address, tags or insurance "
             + "policies respectively.\n"
             + "Use '&' to find for multiple search terms.\n"
+            + "Specify attributes by typing '-[ATTRIBUTE] after keywords\n"
             + "Keywords cannot be empty.\n"
-            + "Parameters: FLAG/KEYWORD [& MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " e/alice@mail.com & bob@mail.com";
+            + "Parameters: FLAG/KEYWORD [& MORE_KEYWORDS]... [-ATTRIBUTE]... (attributes must be policy, "
+            + "phone or email)\n"
+            + "Example: " + COMMAND_WORD + " e/alice@mail.com & bob@mail.com -address -policy";
 
     private final Predicate<Person> predicate;
 
+    private final List<Attribute> attributes;
+
+    /**
+     * Create a FindCommand that lists persons that satisfy given {@code Predicate}
+     * @param predicate by which to filter list
+     */
     public FindCommand(Predicate<Person> predicate) {
         this.predicate = predicate;
+        this.attributes = new ArrayList<>();
+    }
+
+    /**
+     * Create a FindCommand that lists {@code Attribute} of persons that satisfy given {@code Predicate}
+     * @param predicate by which to filter list
+     * @param attributes that find will show
+     */
+    public FindCommand(Predicate<Person> predicate, List<Attribute> attributes) {
+        this.predicate = predicate;
+        this.attributes = attributes;
+    }
+
+    public boolean isAttributeSpecified() {
+        return !this.attributes.isEmpty();
+    }
+
+    private String getAttributesString() throws CommandException {
+        StringBuilder attributeName = new StringBuilder();
+        boolean isFirst = true;
```

Comment:
Suggestion to name this isFirst to isFirstAttribute to make it more understandable.

Reply:
Good point, done!